staging/src/k8s.io/component-base/pkg/feature/feature_gate.go

```diff
@@ -52,7 +52,7 @@ var (
 	}
 
 	// DefaultMutableFeatureGate is a mutable version of DefaultFeatureGate.
-	// Only top-level commands/options setup and the k8s.io/apiserver/pkg/util/feature/testing package should make use of this.
+	// Only top-level commands/options setup and the k8s.io/component-base/pkg/feature/testing package should make use of this.
 	// Tests that need to modify feature gates for the duration of their test should use:
 	//   defer utilfeaturetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.<FeatureName>, <value>)()
 	DefaultMutableFeatureGate MutableFeatureGate = NewFeatureGate()
```

Comment:
this must not move. We have a techinical debt here that we have this global variable at all. Component-base should only have the library code.

Reply:
Also keeping this here will make the PR much smaller.

Reply:
Done! I moved everything except for `DefaultMutableFeatureGate` and `DefaulteFeatureGate` to `k8s.io/component-base/featuregate`.